### PR TITLE
docs(dbt): add reference for creating dbt manifest at run time and build time

### PR DESCRIPTION
**Pull request by rexledesma**
## Summary & Motivation
Adds documentation about:

- why we need a `manifest.json`
- the different options for creating a `manifest.json`
- how these options show up in `dagster-dbt project scaffold`
- our recommendations for development and production 

This removes the last references to `load_assets_from_dbt_project/manifest`. Instead, they've been moved to the `Relevant API's` section, with references to their alternatives.

## How I Tested These Changes
read

### CODE REVIEW

**Comment by rexledesma** [comment]
Current dependencies on/for this PR:
* master
  * **PR #16040** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/16040" target="_blank">Graphite</a> 
    * **PR #16043** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/16043" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/16043?utm_source=stack-comment).

File: docs/content/integrations/dbt/reference.mdx [hunk 1]
```diff
@@ -77,36 +73,45 @@ The `dagster-dbt` library offers two methods of loading dbt models from a projec
...
-from dagster_dbt import load_assets_from_dbt_project
+The manifest can be created in two ways:
 
-dbt_assets = load_assets_from_dbt_project(project_dir="path/to/dbt/project")
-```
+1. **At run time**: A dbt manifest is generated when your Dagster definitions are loaded, or
+2. **At build time**: A dbt manifest is generated before loading your Dagster definitions and is included as part of your Python package.
 
-The <PyObject module="dagster_dbt" object="load_assets_from_dbt_project" /> function:
+In the Dagster project created by the [`dagster-dbt project scaffold`](/\_apidocs/libraries/dagster-dbt#dagster-dbt-project-scaffold) command line interface, we offer you both ways to load your dbt models, depending on your preference:
```
[the remark this answers is not in the document]
Comment (sryza) [reply docs/content/integrations/dbt/reference.mdx:83]: Ah I see that you cover this a little more at the end. I think it's worth frontloading it though.

File: docs/content/integrations/dbt/reference.mdx [hunk 2]
```diff
@@ -31,6 +31,9 @@ For a step-by-step implementation walkthrough, refer to the [Using dbt with Dags
 | <PyObject module="dagster_dbt" object="get_asset_key_for_model" />                             | A helper method that retrieves the <PyObject module="dagster" object="AssetKey" /> for a dbt model.                                                 |
 | <PyObject module="dagster_dbt" object="get_asset_key_for_source" />                            | A helper method that retrieves the <PyObject module="dagster" object="AssetKey" /> for a dbt source with a singular table.                          |
 | <PyObject module="dagster_dbt" object="get_asset_keys_by_output_name_for_source" />            | A helper method that retrieves the <PyObject module="dagster" object="AssetKey" />'s for a dbt source with multiple tables.                         |
+| <PyObject module="dagster_dbt" object="load_assets_from_dbt_project" />                        | Deprecated in favor of <PyObject module="dagster_dbt" object="dbt_assets" decorator />.                                                             |
```
Comment (sryza) [inline docs/content/integrations/dbt/reference.mdx:34]: I think we could leave this out

File: docs/content/integrations/dbt/reference.mdx (hunk 1 above)
[the remark this answers is not in the document]
Comment (erinkcochran87) [reply docs/content/integrations/dbt/reference.mdx:83]: +1

File: docs/content/integrations/dbt/reference.mdx [hunk 3]
```diff
@@ -77,36 +73,45 @@ The `dagster-dbt` library offers two methods of loading dbt models from a projec
...
-from dagster_dbt import load_assets_from_dbt_manifest
+As the comment explains, the code gives you a choice about how to create this dbt manifest. Based on the `DAGSTER_DBT_PARSE_PROJECT_ON_LOAD` environment variable, either:
 
-with open("path/to/dbt/manifest.json") as f:
-    manifest_json = json.load(f)
+1. **At run time**: This code generates the `manifest.json` for you. This is the easiest option during development because you never need to worry about the file being out-of-date with your dbt project, or
+2. **At build time**: This code leaves it up to you to generate the `manifest.json` file on your own, and this code just reads it.
 
-dbt_assets = load_assets_from_dbt_manifest(manifest_json)
+When developing locally, you can run the following command to generate the manifest at run time when developing locally on your dbt and Dagster project:
```
Comment (erinkcochran87) [inline docs/content/integrations/dbt/reference.mdx:108]: This sentence is a bit repetitive:

> When developing locally,

And:

> when developing locally on your dbt and Dagster project

File: docs/content/integrations/dbt/reference.mdx (hunk 2 above)
Comment (sryza) [reply docs/content/integrations/dbt/reference.mdx:34]: +1 to updating the API docs and then leaving this out

Comment (rexledesma) [reply docs/content/integrations/dbt/reference.mdx]: Could you clarify what change we should make in the API docs for `load_assets_from_dbt_project`? We already recommend `@dbt_assets` there. https://github.com/dagster-io/dagster/blob/28391da99fc62b7996e2a84994ae73029d5e59d2/python_modules/libraries/dagster-dbt/dagster_dbt/asset_defs.py#L503-L509

**Review by PedramNavid — APPROVED** [review]

**Pull request merged**